frontend/src/components/ideas/IdeasBoard.js

```diff
@@ -93,10 +93,27 @@ export default function IdeasBoard({
   };
 
   const handleRemoveComment = (c) => {
-    setIdea({
-      ...idea,
-      comments: [...idea.comments.filter((pc) => pc.id !== c.id)],
-    });
+    // remove a top comment
+    if (c.parent_comment_id === null) {
+      setIdea({
+        ...idea,
+        comments: [...idea.comments.filter((pc) => pc.id !== c.id)],
+      });
+      // remove a reply comment
+    } else {
+      const parentCommentIndex = idea.comments.findIndex(
+        (comment) => comment.id === c.parent_comment_id
+      );
+      const filterOutReplies = [
+        ...idea.comments[parentCommentIndex].replies.filter((pc) => pc.id !== c.id),
+      ];
+      idea.comments[parentCommentIndex].replies = filterOutReplies;
```

Comment:
For editing a state you should always use `setIdea()`. Otherwise the state might get corrupted. 
The best way to do this that I can think of is this: 
```suggestion
      setIdea({...idea, comments: [...idea.comments.filter((pc) => pc.id !== c.id, {...c, c.replies.filter((pc) => pc.id !== c.id)}]}
```

Reply:
Redid this using a temp variable for the idea and only editing it then setting the idea to the temp variable.